src/overview/utils/helpers/OverviewUserSettings.ts

```diff
@@ -9,26 +9,29 @@ type OverviewUserSettings = {
 };
 
 type WelcomeSettings = {
-  hideWelcome: boolean;
+  hideWelcome?: boolean;
   save: (showWelcome: boolean) => void;
   clear: () => void;
 };
 
-const parseOrClean = (key) => {
+// eslint-disable-next-line @typescript-eslint/no-unnecessary-type-parameters
+const parseOrClean = <T>(key: string): T | object => {
   try {
-    return JSON.parse(loadFromLocalStorage(key)) ?? {};
+    const storedValue = loadFromLocalStorage(key) ?? '';
+    return JSON.parse(storedValue) as T;
   } catch (_e) {
     removeFromLocalStorage(key);
+    // eslint-disable-next-line no-console
     console.error(`Removed invalid key [${key}] from local storage`);
   }
-  return {};
+  return {} as T;
 };
 
 const saveRestOrRemoveKey = (key: string, { rest }: Record<string, Record<string, unknown>>) => {
-  if (!Object.keys(rest).length) {
-    removeFromLocalStorage(key);
-  } else {
+  if (Object.keys(rest).length > 0) {
```

Comment:
!isEmpty(Object.keys(rest)) ? saveToLocalStorage(key, JSON.stringify({ ...rest })) : removeFromLocalStorage(key);
or just remove the else and add a return